### ISSUE TITLE
Cursor pagination doesn't work

### ISSUE THREAD

**Issue by PaperPlaneSoftware** [wp-graphql/wp-graphql#1854]
Maybe I'm doing something wrong or maybe this is a problem with wp-graphql-woocommerce but cursor pagination seems broken. I'm using the latest version of wp-graphql (v1.3.5). My query looks something like this:

query GetProducts {
    products(
      first: 4,
      after: ""
    ) {
    	edges {
        node {
          name
        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }

The first time I run this (with an empty or null `endCursor`) I get products back and I get a new `endCursor`. However, if I use the returned `endCursor` by inserting it in `after` and rerun the query I get no results.

Oddly, if I then change the number of products I want returned by setting `first` to some other value it suddenly works again.

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware are you using WPGraphQL for WooCommerce or are `products` a Custom Post Type you registered to show in WPGraphQL on your own? 

Can you provide enough information for us to reproduce? 

What plugins you have active, how many products you have and details about them, etc. Possibly even an export of Products we could import, etc.

The more information we have to reproduce the issue in a local dev environment, the better we can help identify and fix issues.

**Comment by PaperPlaneSoftware** [comment]
@jasonbahl Yes I'm using WPGraphQL for WooCommerce (specifically version 0.7.0). I'll try the latest version (0.8.1 I think) tomorrow to see if that resolve the issue.

Other plugins I have active are WPGraphiQL (1.0.1), WPGraphQL CORS (1.1.0), WPGraphQL JWT (0.4.0), WPGraphQL for ACF (0.4.0) and WPGraphQL Offset Pagination (0.2.0) but it's deactivted.

Are there any specific log files I could dump in here?

**Comment by PaperPlaneSoftware** [comment]
I updated to WPGraphQL-WooCommerce 0.8.1 but the same thing is happening.

I also tried the same thing using posts rather than products and it works fine, so I'm guessing this is an issue specifically with WPGraphQL-WooCommerce.

**Comment by jasonbahl** [comment]
I believe @kidunot89 is working on changing how Product Connections resolve to make more use of some of the internal WPGraphQL connection resolvers. I'm not sure the progress/ETA on that, but that might end up solving this issue when that change is complete. 

I'll transfer this issue to that repo to track over there.
